src/lib/data/token.js

```diff
@@ -1,14 +1,20 @@
 export default class Token {
-  constructor ({ textType, idWord, word, beforeWord, afterWord, hasLineBreak }) {
+  constructor ({ textType, idWord, word, beforeWord, afterWord, hasLineBreak } = {}, segmentIndex, docSourceId) {

```

Comment:
Because it is created from the tokenization service,
we receive token-like object from the service, and pass it as the first parameter, and add aditional parameters from the environment, that were not used for the tokenization - `docSourceId` and `segmentIndex`.
So that's why 
`{ textType, idWord, word, beforeWord, afterWord, hasLineBreak }` - is main token data
`segmentIndex, docSourceId` - is additional data for optimizing perfomance

Reply:
That makes sense. Thanks for explaining!